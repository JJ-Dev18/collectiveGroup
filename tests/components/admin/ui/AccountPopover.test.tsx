import { useRouter } from 'next/router';
import { render, screen, fireEvent } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import AccountPopover from 'fleed/components/admin/ui/adminHeader/AccountPopover';

jest.mock('next/router', () => jest.requireActual('next-router-mock'))



describe('Pruebas con el accountpopover', () => {
  const logout = jest.fn()
  
  test('Debe ser igual al snapshot', () => {
   const { container  } =  render(<AccountPopover logout={logout} />);
   expect(container).toMatchSnapshot()
   
  });

  test('debe abrir el menu al dar click', () => { 
    render(<AccountPopover logout={logout} />);
    fireEvent.click(screen.getByRole('button'))
    expect(screen.getByText('Home'))
    expect(screen.getByText('Logout'))

    screen.debug()
   })

   test('se debe llamar la funcion logout al dar click', () => { 
    render(<AccountPopover logout={logout} />);
    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByRole('menuitem',{ name : 'Logout'}))
    expect(logout).toHaveBeenCalled()
    })


});