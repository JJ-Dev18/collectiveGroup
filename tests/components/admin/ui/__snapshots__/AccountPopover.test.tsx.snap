// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el accountpopover Debe ser igual al snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-3xdcvl-MuiAvatar-root"
    >
      NC
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
